modules/gapi/src/streaming/onevpl/engine/preproc/preproc_engine.cpp

```diff
@@ -0,0 +1,491 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifdef HAVE_ONEVPL
+
+#include <algorithm>
+#include <exception>
+
+#include <opencv2/gapi/streaming/onevpl/data_provider_interface.hpp>
+
+#include "streaming/onevpl/engine/preproc/preproc_engine.hpp"
+#include "streaming/onevpl/engine/preproc/preproc_session.hpp"
+
+#include "streaming/onevpl/accelerators/accel_policy_interface.hpp"
+#include "streaming/onevpl/accelerators/surface/surface.hpp"
+#include "streaming/onevpl/cfg_params_parser.hpp"
+#include "logger.hpp"
+
+#define ALIGN16(value)           (((value + 15) >> 4) << 4)
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+bool FrameInfoComparator::operator()(const mfxFrameInfo& lhs, const mfxFrameInfo& rhs) const {
+    return lhs < rhs;
+}
+
+bool FrameInfoComparator::equal_to(const mfxFrameInfo& lhs, const mfxFrameInfo& rhs) {
+    return lhs == rhs;
+}
```

Comment:
Why do we need this, given that `<` and `==` are available already?

Reply:
in different namespaces:
`std::less` which is required  for `std::map` try to find operator< in global or std namespace and ignored cv::gapi::wip::onevpl. By this reason i use `FrameInfoComparator` in `std::map`
On the other side my UnitTests on preproc requires checking` FrameInf`o before and after preproc, which means i should make operator<, and operator== for FrameInfo visible as public symbols in `utils.hpp` which looks confusing (for me) because utils is not part of interface

On the other side, because `VPPPreprocEngine` is already visible (for UT) purposes and it still require operator< (because ADL doesn't work) then i put aditional operation `equal_to` for this comparator class and made the whole file with VPPPreprocEngine visible for UTs